apps/omg_child_chain/lib/omg_child_chain/fee_parser.ex

```diff
@@ -39,28 +39,75 @@ defmodule OMG.ChildChain.FeeParser do
     end
   end
 
-  defp parse_fee_spec(%{"flat_fee" => fee, "token" => token}) do
+  defp parse_fee_spec(%{
+         "amount" => fee,
+         "token" => token,
+         "pegged_amount" => pegged_amount,
+         "pegged_currency" => pegged_currency,
+         "pegged_subunit_to_unit" => pegged_subunit_to_unit,
+         "updated_at" => updated_at
+       }) do
     # defensive code against user input
-    with {:ok, fee} <- validate_fee(fee),
-         {:ok, addr} <- decode_address(token) do
-      %{token: addr, flat_fee: fee}
+    with {:ok, fee} <- validate_positive_amount(fee, :invalid_fee),
+         {:ok, addr} <- decode_address(token),
+         {:ok, pegged_amount} <- validate_positive_amount(pegged_amount, :invalid_pegged_amount),
+         {:ok, pegged_currency} <- validate_pegged_currency(pegged_currency),
+         {:ok, pegged_subunit_to_unit} <-
+           validate_positive_amount(pegged_subunit_to_unit, :invalid_pegged_subunit_to_unit),
```

Comment:
Very slim chance but possible that `pegged_subunit_to_unit = 0` which passes the validation and cause divided by zero later on.

Also, very nitpicky... might need to differentiate between `validate_positive_amount/2` and `validate_non_negative_amount/2`, currently it's closer to the latter but named as the former.